### PR TITLE
Upgrade to V3

### PR DESCRIPTION
**Pull request by jhaaaa**
- Adds doc about upgrading to XMTP V3
- Preview addition to [left nav and home page](https://docs-xmtp-org-git-upgrade-v3-ephemerahq.vercel.app/)
- Preview [Upgrade to XMTP V3](https://docs-xmtp-org-git-upgrade-v3-ephemerahq.vercel.app/upgrade-to-v3)
   - Thank you for the "Key Takeaways" callout pattern, @saulmc!

### CODE REVIEW

File: docs/pages/upgrade-to-v3.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+- In V1 and V2, we used intros V1 (`fromIntro`) and invites V2 (`fromInvite`) to decrypt invites and intros.
+- In V3, new conversations are called welcomes, and we use `fromWelcome` to decrypt them.
+
+### Getting topic IDs
+
+- In V2, we used `/xmtp/0/invite-$address/proto` to get the topic identifier for a new conversation.
+- In V3, we use `/xmtp/mls/1/w-$installationId/proto` to get the topic identifier for a new group chat or DM conversation.
+
+## Installation-specific private keys
+
```
Comment (jhaaaa) [inline docs/pages/upgrade-to-v3.md:42]: @nplasterer Did I capture what you shared about v2 vs v3 identity accurately here? <3

File: docs/pages/upgrade-to-v3.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+In V2, we had a `privateKeyBundle` that could be exported and imported into any SDK to create a client. This allowed users to migrate their identity across SDKs without re-signing with their wallet. In V2, we also had a consistent identity key that we used to sign and verify messages, and their static nature could make them a vulnerability over time.
+
+In V3, we have installation-specific key bundles that stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that creating a V3 client is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database.
```
Comment (nplasterer) [inline docs/pages/upgrade-to-v3.md:44]:
```suggestion
In V3, we have installation-specific key bundles that are stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that after a V3 client is created it is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database. With the added security of installation based keys it is no longer possible to export and import them into other clients. Calling create will create a new installation if a local database is not already present. However you inboxId should be the same across all installations.
```

File: docs/pages/upgrade-to-v3.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+- In V3, we use `/xmtp/mls/1/w-$installationId/proto` to get the topic identifier for a new group chat or DM conversation.
+
+## Installation-specific private keys
+
+In V2, we had a `privateKeyBundle` that could be exported and imported into any SDK to create a client. This allowed users to migrate their identity across SDKs without re-signing with their wallet. In V2, we also had a consistent identity key that we used to sign and verify messages, and their static nature could make them a vulnerability over time.
+
+In V3, we have installation-specific key bundles that are stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that after a V3 client is created it is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database. With the added security of installation based keys it is no longer possible to export and import them into other clients. Calling create will create a new installation if a local database is not already present. However your inboxId should be the same across all installations.
```
Comment (nplasterer) [inline docs/pages/upgrade-to-v3.md]:
```suggestion
In V3, we have installation-specific key bundles that are stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that after a V3 client is created it is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database. With the added security of installation based keys it is no longer possible to export and import them into other clients. Calling create will create a new installation if a local database is not already present. Provided the same wallet is used for signing, your `inboxID` will be the same across all installations.
```

File: docs/pages/upgrade-to-v3.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+- In V3, we use `/xmtp/mls/1/w-$installationId/proto` to get the topic identifier for a new group chat or DM conversation.
+
+## Installation-specific private keys
+
+In V2, we had a `privateKeyBundle` that could be exported and imported into any SDK to create a client. This allowed users to migrate their identity across SDKs without re-signing with their wallet. In V2, we also had a consistent identity key that we used to sign and verify messages, and their static nature could make them a vulnerability over time.
+
+In V3, we have installation-specific key bundles that are stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that after a V3 client is created, it is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database. With the added security of installation-based keys, it is no longer possible to export and import them into other clients. Calling create will create a new installation if a local database is not already present. However, your inbox ID should be the same across all installations.
```
Comment (nplasterer) [inline docs/pages/upgrade-to-v3.md:44]:
```suggestion
In V3, we have installation-specific key bundles that are stored securely in the managed local database and rotated periodically, which adds a significant layer of security. This means that after a V3 client is created, it is as simple as calling the `build` method with an address and a `dbEncryptionKey`, which retrieves the appropriate key bundle from the local database. With the added security of installation-based keys, it is no longer possible to export and import them into other clients. Calling create will create a new installation if a local database is not already present. Provided the same wallet is used for signing, your `inboxID` will be the same across all installations.
```

**Review by nplasterer — APPROVED** [review]

**Pull request merged**